File: Resueltos_por_competidor/Jacobo_Segura/Police_Recruits.java
/* 
Codeforces 427A - Police Recruits
https://codeforces.com/problemset/problem/427/A
*/
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int pol = 0;
        int sa = 0;
        
        while(n-- > 0) {
        	int g = sc.nextInt();
        	if(g == -1) {
        		if(pol>0) {
        			pol--;
        		}else {
        			sa++;
        		}
        	}else {
        		pol += g;
        	}
        }
        System.out.println(sa);
    }
}

File: Resueltos_por_competidor/Juan_Martinez/PoliceRecruits.cpp
/*
 * Autor: Juan Martinez
 * Problema: Police Recruits (427A)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/427/A
 * Difficulty: 800
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, aux; cin>>n;
    int policias = 0, crimes = 0;
    for(int i = 0; i < n; i++) {
        cin>>aux;
        if(aux == -1) {
            if(policias > 0) policias--;
            else crimes++;
        } else policias += aux;
    }
    cout<<crimes<<endl;
}
